python/perf-kernels/streamk/utils/file_generator.py

```diff
@@ -0,0 +1,377 @@
+    3. test_gemm to invoke matmul in a loop of {iters} iterations
+    """
+
+    filenames = []
+    for i in range(jobs):
+        filenames.append(get_filename_profile_driver(M, N, K, i))
+    f_kernel = [open(path, 'w') for path in filenames]
+
+    # write imports
+    import_str = """import torch
+import triton
+import triton.language as tl
+import argparse
+import sys
+import multiprocessing
+from tune_streamk import gen_rotating_tensors
+"""
+    if icache_flush:
+        import_str += """
+from icache_flush import icache_flush
+"""
+    for fi in range(jobs):
+        f_kernel[fi].write(import_str + "\n")
+
+    with open("streamk_kernel.py") as file:
+        streamk_gemm_code = file.read()
+
+    idx = 0
+    for config in configs:
+        file_idx = idx % jobs
+        EVEN_K = True if K % config.get('BLOCK_SIZE_K') == 0 else False
+        configStr, matmul_def_str = gen_kernel_and_configStr_from_config(config, EVEN_K, dtype_a, dtype_b, dtype_c,
+                                                                         dtype_p, dtype_lock, bias_size, False)
+        # Copy the streamk_gemm with name replaced
+        streamk_gemm_config = streamk_gemm_code.replace("streamk_gemm", f"streamk_gemm_{configStr}")
+        streamk_gemm_config = streamk_gemm_config.replace("import triton.language as tl", "")
+        streamk_gemm_config = streamk_gemm_config.replace("import triton", "")
+        f_kernel[file_idx].write(streamk_gemm_config + "\n\n")
+        # Copy the matmul_kernel with name replaced
+        f_kernel[file_idx].write(matmul_def_str + "\n")
+        idx += 1
+
+    # write test_gemm
+    # pre string
+    test_gemm_pre_str = f"""def test_gemm(M, N, K, rotating_buffer_size, bias_size):
+    tensors = gen_rotating_tensors(M, N, K, '{dtype_a}', {col_a}, '{dtype_b}', {col_b}, '{dtype_c}',
+                                   1, '{init_type}', rotating_buffer_size, bias_size, device='cuda')
+
+    a = tensors['input_a'][0]
+    b = tensors['input_b'][0]
+    c = tensors['output_c'][0]
+    assert bias_size == M or bias_size == 0
+
+    stride_bias = tensors['bias'][0].stride(0) if bias_size > 0 else 0
+
+    try:
+        with open("{get_filename_compile_driver()}.failed_configs", "r") as f:
+            failed_configs = [cfg.strip() for cfg in f.readlines()]
+    except Exception:
+        failed_configs = []
+"""
+    for fi in range(jobs):
+        f_kernel[fi].write(test_gemm_pre_str + "\n")
+
+    # call all matmul_xxx functions
+    idx = 0
+    runs = iters if run_bench else 120
+    for config in configs:
+        configStr = gen_configStr(config)
+        block_m = config.get('BLOCK_SIZE_M')
+        block_n = config.get('BLOCK_SIZE_N')
+        num_sms = config.get('NUM_SMS')
+        matmul_call_str = f"""
+    if '{configStr}' not in failed_configs:
+        rotating_num = tensors['rotating_num']
+        locks = torch.zeros(({runs}, {num_sms}), device = "cuda", dtype = torch.int32)
```

Comment:
`locks` can be less than `int32` type, we only need 1 byte: `uint8` should work. Tensile uses `uint8`.